modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/FileEntryPicker.js

```diff
@@ -22,20 +22,39 @@ import React, {useEffect, useState} from 'react';
 
 const FileNamePicker = ({
 	maxFileSize: initialMaxFileSize,
+	maxMimeTypeSize,
 	namespace,
 	validExtensions,
 }) => {
-	const maxFileSize = Number(initialMaxFileSize);
+	let maxFileSize = Number(initialMaxFileSize);
 	const inputId = `${namespace}file`;
 	const [inputValue, setInputValue] = useState('');
 	const [fileName, setFileName] = useState('');
 	const [maxFileSizeError, setMaxFileSizeError] = useState(false);
+	const [maxSize, setMaxSizeValue] = useState(maxFileSize);
 
 	useEffect(() => {
 		setFileName(inputValue ? inputValue.replace(/^.*[\\]/, '') : '');
 	}, [inputValue]);
 
 	const onInputChange = ({target}) => {
+		const jsonObj = JSON.parse(maxMimeTypeSize);
+		const type = target.files[0].type;
+		const maxMimeSize = Number(jsonObj[type]);
+		maxFileSize = Number(initialMaxFileSize);
+
+		if (
+			Object.prototype.hasOwnProperty.call(
+				jsonObj,
+				target.files[0].type
+			) &&
+			maxFileSize > maxMimeSize
+		) {
+			maxFileSize = maxMimeSize;
+		}
+
+		setMaxSizeValue(maxFileSize);
+
```

Comment:
Can you try to simplify the conditions and the code around, taking into account the previous comments, something like this: (_*pseudocode*_)

```suggestion
		const fileType = target.files[0]?.type;
		const maxFileTypeSize = Number(maxMimeTypeSize[type]);

		if (
			maxFileTypeSize &&
			maxFileSize > maxFileTypeSize
		) {
			setMaxSizeValue(maxFileTypeSize);
		} else {
		        setMaxSizeValue(Number(initialMaxFileSize));
		}
```

Reply:
I applied it in my new code lines